### PR TITLE
DEV2-2461: Create a thin plugin for IJ for publishing

### PR DESCRIPTION
**Pull request by assapir**

### CODE REVIEW

**Review by amircodota — APPROVED** [review]
LGTM

File: Common/src/main/java/com/tabnineCommon/capabilities/CapabilitiesService.java [hunk 1]
```diff
@@ -30,9 +30,7 @@ public static CapabilitiesService getInstance() {
   }
 
   public void init() {
-    if (!Config.IS_SELF_HOSTED) {
-      scheduleFetchCapabilitiesTask();
-    }
+    scheduleFetchCapabilitiesTask();
```
Comment (amircodota) [inline Common/src/main/java/com/tabnineCommon/capabilities/CapabilitiesService.java:33]: why?

File: TabnineSelfHosted/build.gradle [hunk 2]
```diff
@@ -77,4 +78,11 @@ runPluginVerifier {
     if (project.hasProperty("localPaths")) {
         localPaths = Arrays.stream(project.localPaths.split(',')).map(File::new).collect(Collectors.toList())
     }
+}
+
+sourceSets {
+    main {
+        java.srcDirs += "../TabnineSelfHostedForMarketplace/src/main/java"
+        java.srcDirs += "../channels/self_hosted"
```
Comment (amircodota) [inline TabnineSelfHosted/build.gradle:86]: why did you move thse files?

**Review by amircodota — COMMENTED** [review]
see my comments

File: Common/src/main/java/com/tabnineCommon/capabilities/CapabilitiesService.java (hunk 1 above)
Comment (assapir) [reply Common/src/main/java/com/tabnineCommon/capabilities/CapabilitiesService.java:33]: it won't be called anyway when self-hosted.

**Review by amircodota — APPROVED** [review]
LGTM
